pkg/kind/kong_proxy.go

```diff
@@ -109,7 +97,17 @@ func (c *kongProxyCluster) startDeploymentInformer(ctx context.Context) (deploym
 	return
 }
 
-func (c *kongProxyCluster) startServiceInformer(ctx context.Context, nsn types.NamespacedName, ready chan ProxyReadinessEvent) {
+// startServiceInformer exposes the provided deployment as LoadBalancer type and reports when the service has been provisioned a load balancer IP/Host
+func (c *kongProxyCluster) startServiceInformer(ctx context.Context, d *appsv1.Deployment, ready chan ProxyReadinessEvent) {
+	// expose the deployment first, if this fails then all fails
+	_, err := c.exposeProxyDeployment(ctx, d)
+	if err != nil {
+		ready <- ProxyReadinessEvent{Err: err}
+		close(ready)
```

Comment:
Might be better to `defer close(ready)` at the beginning of this function instead of `close(ready)` before every `return`.

Why?
- This way we're guaranteed not to overlook channel closing
- less code
- `defer` will work also if this fn ever `panic`s

Reply:
The problem is that there are conditions where this function should return without closing, the explicit closes cover the conditions where we should close and be done.

Reply:
Sounds good, will implement.

Reply:
:+1: bbfeaba1e62ab5036a0e198710a4559dd1ab1070